### PR TITLE
added more flexible binning

### PR DESCRIPTION
**Pull request by fjaviersanchez**
Some of us have discussed online the possibility of adding more flexible binning for the covariance calculation using TJPCov, for example, reading bandpower windows. Here's a first draft implementation for that.

### CODE REVIEW

**Comment by fjaviersanchez** [comment]
@carlosggarcia, please feel free to assign someone else as a reviewer!!

File: tjpcov/covariance_gaussian_fsky.py [hunk 1]
```diff
@@ -41,24 +42,37 @@ def get_binning_info(self, binning="linear"):
...
-            del_ell = (ell_eff[1:] - ell_eff[:-1])[0]
-
-            ell_min = ellb_min - del_ell / 2
-            ell_max = ellb_max + del_ell / 2
-
-            ell_delta = (ell_max - ell_min) // nbpw
-            ell_edges = np.arange(ell_min, ell_max + 1, ell_delta)
-            ell = np.arange(ell_min, ell_max + ell_delta - 2)
+        # https://github.com/LSSTDESC/TXPipe/blob/a9dfdb7809ac7ed6c162fd3930c643a67a
+        out = super().get_binning_info()
```
Comment (carlosggarcia) [inline tjpcov/covariance_gaussian_fsky.py:46]: Why not `self.get_binning_info()`?

**Review by carlosggarcia — CHANGES_REQUESTED** [review]
Seems to be some problem with the unit tests, I think it has to do with GitHub because they passed in my laptop. For some reason, I cannot rerun the actions, so cannot check.

Just a minor comment.

File: tjpcov/covariance_gaussian_fsky.py (hunk 1 above)
Comment (fjaviersanchez) [reply tjpcov/covariance_gaussian_fsky.py:46]: Updated!

**Review by carlosggarcia — APPROVED** [review]
LGTM!

**Comment by fjaviersanchez** [comment]
thanks! I think that the unit tests are broken. I can open a separate PR to try to fix them and then rebase because it doesn't allow me to merge as is :(

**Comment by carlosggarcia** [comment]
Actually it looks like that! Pinging @damonge. I think we can merge this since the tests passed in my laptop

**Pull request merged**